### PR TITLE
Scaffolding for the student form for entering hours.

### PR DESCRIPTION
**Pull request by kosigz**
Changing the main Home.js thingy to have the student interface.

### CODE REVIEW

File: TextBox.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.input {
+  width: 100%;
```
Comment (deadlocked247) [inline TextBox.css:2]: We don't need this, use the semantic-ui input class

```
<div class="ui fluid icon input">
  <input type="text" placeholder="Search a very wide input...">
  <i class="search icon"></i>
</div>
```

File: src/components/Dropdown.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React, { Component, PropTypes } from 'react'
+
+class Dropdown extends Component {
+  static propTypes = {
+    children: PropTypes.any,
+    title: PropTypes.string,
+    options: PropTypes.array,
+    isVisible: PropTypes.bool,
+    selected: PropTypes.string,
```
Comment (deadlocked247) [inline src/components/Dropdown.js:9]: Can we also make this `PropTypes.string.isRequired` so the user of this component must provide a selected string?

File: src/components/Dropdown.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    onClick: PropTypes.func,
+  }
+  static defaultProps = {
+    onClick: (item) => { console.log(item) },
```
Comment (deadlocked247) [inline src/components/Dropdown.js:13]: Let's just make this `console.debug(item)` so the user will not see it in the console

File: src/components/Dropdown.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    options: ['male', 'female'],
+    selected: ''
+  }
+  render() {
+    const { selected, children, options, title, onClick } = this.props
+    return (
+      <div>
+      {title}
+      <div className="ui fluid simple dropdown">
+        <input type="hidden" name="gender" />
```
Comment (deadlocked247) [inline src/components/Dropdown.js:23]: don't need this `name` here, this should be a prop passed down

File: src/components/Dropdown.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        <i className="dropdown icon"></i>
+        {selected === '' ?
+          <div className="default text">{'Select ' + title + ' from the dropdown!'}</div>
```
Comment (deadlocked247) [inline src/components/Dropdown.js:26]: The text that appears when there is no text should be a prop

Comment (deadlocked247) [reply src/components/Dropdown.js:26]: Just add that `'Selected' + title ' from the dropdown!'` as the default prop

File: src/components/Dropdown.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+          : null
+        }
+        {selected}
+        <div className="menu fluid">
+          <div className="item fluid" onClick={() => onClick(options[0])}>{options[0]}</div>
+          <div className="item fluid" onClick={() => onClick(options[1])}>{options[1]}</div>
```
Comment (deadlocked247) [inline src/components/Dropdown.js:32]: This needs to be a `map` that goes over all the possible selectable inputs (given by the user), so something like

```
{
    this.props.possibleChoices.map(choice => {
        return (<div ....  onClick={() => this.props.onClick(choice)} >....)
    })
}
```

File: src/components/Modal.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    children: PropTypes.any,
+    header: PropTypes.string,
+    approveLabel: PropTypes.string,
+    cancelLabel: PropTypes.string,
+    approveOnClick: PropTypes.func,
+    cancelOnClick: PropTypes.func,
+    isVisible: PropTypes.bool,
+  }
+  static defaultProps = {
+    header: 'Modal!',
```
Comment (deadlocked247) [inline src/components/Modal.js:16]: Just `Modal`

File: src/components/Modal.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    approveLabel: 'Submit',
+    cancelLabel: 'Cancel',
+    approveOnClick: () => { console.log('yes') },
+    cancelOnClick: () => { console.log('no') },
```
Comment (deadlocked247) [inline src/components/Modal.js:20]: `console.debug`

File: src/components/Modal.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    isVisible: false,
+  }
+  render() {
+    const { className, label, onClick, children, header, isVisible, approveOnClick, approveLabel, cancelLabel, cancelOnClick } = this.props
```
Comment (deadlocked247) [inline src/components/Modal.js:24]: once this gets this big, just break up into lines for legibility

File: src/components/Modal.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    if (!isVisible) {
+      console.log('is not visible.')
```
Comment (deadlocked247) [inline src/components/Modal.js:26]: dont need this log, this will clutter the user's console

File: src/components/Modal.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+      return null;
+    }
+    console.log('is visible!')
```
Comment (deadlocked247) [inline src/components/Modal.js:29]: remove this

File: src/components/Modal.js [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    return (
+      <div className = 'backdrop'>
```
Comment (deadlocked247) [inline src/components/Modal.js:31]: extra spaces

File: src/components/Modal.js [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+        <div className='ui modal active'>
+          <div className='header'>
+            {header}
+          </div>
+          <div className='content'>
+            {console.log(children)}
```
Comment (deadlocked247) [inline src/components/Modal.js:37]: dont log this lol

File: src/components/Modal.js [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+            {children}
+
+          </div>
+          <div className='actions'>
+            <div className='ui approve button' onClick={approveOnClick}>{approveLabel}</div>
```
Comment (deadlocked247) [inline src/components/Modal.js:42]: you should render these `divs` ONLY if there is the associated label, so

```
{approveLabel ? <div ... > : null}
```

File: src/components/StudentForm.js [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+  static propTypes = {
+    className: PropTypes.string,
+    children: PropTypes.any,
+    label: PropTypes.string,
+    onClick: PropTypes.func,
+  }
+  render() {
+    return (
+      <div>
+        WHY?!?!?!?
```
Comment (deadlocked247) [inline src/components/StudentForm.js:22]: ?

File: src/components/StudentForm.js [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+        <Dropdown title='Type of Service'/>
+        <hr/>
+        <Dropdown title='Service Partner'/>
+        <hr/>
+        <Dropdown title='Service Project'/>
```
Comment (deadlocked247) [inline src/components/StudentForm.js:27]: looks good, just pass props into the `Dropdown` component you worked so hard on!

File: src/components/TextBox.js [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+    children: PropTypes.any,
+    title: PropTypes.string,
+    options: PropTypes.array,
+    placeholderText: PropTypes.string,
+  }
+  render() {
+    const { children, placeholderText } = this.props
+    return (
+      <div className='modal'>
+        ???
```
Comment (deadlocked247) [inline src/components/TextBox.js:14]: ???

File: src/components/TextBox.js [hunk 18]
```diff
@@ -0,0 +1,40 @@
+import React, { Component, PropTypes } from 'react'
+
+class TextBox extends Component {
+  static propTypes = {
+    children: PropTypes.any,
+    title: PropTypes.string,
+    options: PropTypes.array,
```
Comment (deadlocked247) [inline src/components/TextBox.js:7]: Use `arrayOf` here to define what's in the array

File: src/components/TextBox.js [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+    const { children, placeholderText } = this.props
+    return (
+      <div className='modal'>
+        ???
+        <br/>
+        Start Time
+        <div className="ui calendar eight wide column" id="example5">
+          <div className="ui input left icon">
+            <i className="calendar icon"></i>
+            <input type="text" placeholder="Date" value="5/30/2015 3pm" />
```
Comment (deadlocked247) [inline src/components/TextBox.js:20]: Yea a lot of attributes and values in this component can be abstracted, or else its pretty useless

File: src/pages/Home.js [hunk 20]
```diff
@@ -77,20 +81,20 @@ class Home extends Component {
     const { location } = this.state
 		return (
 			<div className="ui container">
+        <Modal isVisible={this.state.isModalVisble} approveOnClick={() => {this.setState({isModalVisble: false})}} cancelOnClick={() => {this.setState({isModalVisble: false})}} header='Enter Hours'>
+          <StudentForm></StudentForm>
+        </Modal>
 				<div className="ui three column stackable grid">
           <div className="sixteen wide column">
-            <Map style={{ height: '400px', borderRadius: '5px' }} defaultCenter={location} />
+          <Button className='ui button fluid' onClick={() => {this.setState({isModalVisble: true})}}>
```
Comment (deadlocked247) [inline src/pages/Home.js:89]: probably just pull this out into a diff function

File: test/components/Dropdown.test.js [hunk 21]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { mount } from 'enzyme'
+import { spy } from 'sinon'
+
+import Dropdown from '../../src/components/Dropdown'
+
+describe('Dropdown', () => {
```
Comment (deadlocked247) [inline test/components/Dropdown.test.js:8]: NICE

File: test/components/Modal.test.js [hunk 22]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { mount } from 'enzyme'
+import { spy } from 'sinon'
+
+import Modal from '../../src/components/Modal'
+
+describe('Modal', () => {
+	it('renders without crashing', () => {
```
Comment (deadlocked247) [inline test/components/Modal.test.js:9]: double nice

**Review by deadlocked247 — CHANGES_REQUESTED** [review]
Great start but when you make reusable components (in general not just react), you want to abstract away as much as possible. And understand the use case for each component

File: test/components/Modal.test.js (hunk 22 above)
Comment (tonyjhuang) [reply test/components/Modal.test.js:9]: double plus nice

File: src/components/Dropdown.js (hunk 2 above)
Comment (kosigz) [reply src/components/Dropdown.js:9]: The clients wanted to have no default values for the dropdowns.

File: src/components/Dropdown.js (hunk 3 above)
Comment (kosigz) [reply src/components/Dropdown.js:13]: Sound good to me!

**Pull request closed without merging**